pages/docs/overlay/modal.mdx

```diff
@@ -446,6 +446,85 @@ there are certain situations where you might not want the modal to trap focus.
 
 To prevent focus trapping, pass `trapFocus` and set its value to `false`.
 
+### Styling the backdrop
+
+The backdrop's background by default is set to `blackAlpha.600`, but if you want
+to achieve a different style you can also use the backdrop style props, like
+`backdropBlur`, `backdropBrightness`, `backdropContrast`, `backdropHueRotate`,
+`backdropInvert`, and `backdropSaturate`. To use these style props, you'd have
+to set the `backdropFilter` prop to `auto`.
+
+> Please be aware that not every browser supports the
+> [backdrop-filter](https://caniuse.com/css-backdrop-filter) CSS property, the
+> example below included.
+
+The `filter` property will have no effect on the background because the Modal is
+rendered within a `Portal`. This mean you can only style components within the
+Modal by using this property.
+
+```jsx
+function BackdropExample() {
+  const { isOpen, onOpen, onClose } = useDisclosure()
+  const [overlay, setOverlay] = React.useState()
```

Comment:
Since we're doing TypeScript mostly, the `useState` without a type will break. Take a look here: https://codesandbox.io/s/awesome-babycat-6o63l?file=/src/index.tsx. Maybe you're using JavaScript and everything seems okay, but we do need to provide the type `<JSX.Element>` to make this work.

Reply:
Dont get me wrong, I prefer TypeScript as well. Main reason for me was that there are much more jsx examples than tsx and there is no example with `JSX.Element` as type for anything.

I wonder about your Codesandbox because it work properly for me, or did I miss anything?

Reply:
That's weird. Here's how it looks like in my case:
![CleanShot 2022-02-15 at 09 40 12](https://user-images.githubusercontent.com/5396211/154024540-2fd1bda2-0d45-4a55-99f5-ffad6d2144f1.png)

When you don't provide a type or an initial value, TypeScript has no way of knowing which type the state variable will be. Instead of setting a type, maybe you could set the initial value to `<OverlayOne />`, so the type can be inferred from it.

Reply:
That's it! If you want to clean it up, move the `OverlayOne` and `OverlayTwo` out of the `BackdropExample` component. They are component anyways.

Reply:
Know exactly what you mean. Tried that as well and failed because of that Live-Code thing... :disappointed: 

Only example I found with multiple components is this: https://chakra-ui.com/docs/form/editable#with-custom-controls
This is basically what I have done in my CSB.

Used that code from the CSB now and cleaned it up a bit. If you have any further suggestions let me know!